codegen-library.sh

```diff
@@ -133,16 +133,18 @@ function restore-changes-if-its-copyright-year-only() {
   local difflist
   log "Cleaning up generated code"
   difflist="$(mktemp)"
-  git diff --exit-code --name-only > "$difflist"
```

Comment:
I guess this was killing the script?

Reply:
Yep, it was exiting with `1` not continuing to actual content check. Perhaps we could narrow down the conditions, i.e. only check boilerplate copyright in `zz_generated*.go` files. I'm not sure if it behaves differently in various Bash instances, but wrapping it into the if should be safer for now.

Reply:
In shell, when you set `exit at error` flag (which you should), every command that returns non 0 will break the script.

Unless it is used in conditional statement, like: `if` or combined with `||` or `&&`.

Reply:
Yes, this is a bug. And the fix looks good to me. Thanks!